src/main/java/com/smartagilify/core/entities/BaseEntity.java

```diff
@@ -34,4 +31,16 @@ public abstract class BaseEntity extends MainBaseEntity {
     protected Integer rowLevelId;
     @Column(name = HibernateStaticValues.PRIORITY)
     protected Integer priority;
+
+    @PrePersist
+    protected void prePersistEntity() {
+        this.state = EN_STATE.CREATED;
+        this.createDate = LocalDateTime.now();
+    }
+
+    @PreUpdate
+    protected void preUpdateEntity() {
+        if (this.state == null) this.state = EN_STATE.UPDATED;
```

Comment:
Shouldn't it be this.state != null, the way I see it is if state is null then the entity is not saved yet